### ISSUE TITLE
No longer detecting GitHub releases

### ISSUE THREAD

**Issue by TylerJohnsonDev** [YahnisElsts/plugin-update-checker#287]
I've been using this library for several years without issue. All of the plugins I set up to use the library are no longer updating when an update is pushed out. Any idea what's going on? Thanks!

**Comment by YahnisElsts** [comment]
I just tried it with a couple of test projects and GitHub API still seems to be working fine for me. Has anything changed in your repository settings recently? Maybe the access token has been deleted or it no longer has the required permissions?

To help with debugging, I would recommend installing [Debug Bar](https://wordpress.org/plugins/debug-bar/). Go to the "PUC (your-slug)" tab in the Debug Bar and try clicking the "Check Now" button. If there are any API errors, they should show up there.

**Comment by gregbast1994** [comment]
Is there still a check now button? I do not see one.

**Comment by YahnisElsts** [comment]
Yes, it still shows up for me. Here's an example with version 4.7 of plugin-update-checker:

![image](https://user-images.githubusercontent.com/2527434/64710953-882df280-d4c1-11e9-9e68-788ad36d237b.png)

Note that the button might only show up when you open the debug bar in the admin panel, not in the site front-end.

There's also a "check for updates" link under the plugin description:
![image](https://user-images.githubusercontent.com/2527434/64710859-5a48ae00-d4c1-11e9-9a8d-11f645370b31.png)

**Comment by odil-io** [comment]
Hi, i experienced the same issue out of no where and tried this approach.
But the "Check Now" button doesn't do anything. I even checked network activity on both the server and in browser and really nothing happens.

What I'm trying to accomplish is a more faster update checking schedule or a way for me to Check for a new version when ever I want. I also could not find any info (besides going trough code) on how to "force" an check.

**Comment by gregbast1994** [comment]
@odil-io 

Here is the code I am using:

```
require 'plugin-update-checker/plugin-update-checker.php';
$myUpdateChecker = Puc_v4_Factory::buildUpdateChecker(
	'https://github.com/gregbast1994/msp-storefront', //path
	__FILE__,
	'msp-storefront' // name of the folder which houses your theme/plugin
);

$myUpdateChecker->setBranch('master');
$myUpdateChecker->getVcsApi()->enableReleaseAssets();
```

Every time I want to update, I make sure to bump the version number in my project.

Also, I make sure that the 3rd parameter I pass to  `buildUpdateChecker()` is an EXACT match to the name of the directory that houses my project.

Then I go to GitHub and create a new release titled something like 'v1.74'.

Finally, with the debug bar plugin enabled, navigate to the top of wp-admin (the backend) and click: Debug > PUC > Check Now. Navigate to Themes and click update.

Sometimes for an unknown reason the update MIGHT fail, I wait 10 seconds, refresh (CTRL+R) and try again... it usually works eventually.

I don't know how many of these steps are necessary; but I tried to provide a thorough answer because **this plugin really upped my WP development game** and I want it to work for you (and everyone else too!).

**Comment by YahnisElsts** [comment]
@odil-io 
The "Check Now" button is implemented using a JavaScript event handler, so maybe there's a JS error somewhere that stops it from working? Take a look at the browser's JS console and see if any errors show up when you load the page or when you click the button.

The easiest way to change the update schedule is to pass the desired check period to `buildUpdateChecker()`. For example:

```php
$myUpdateChecker = Puc_v4_Factory::buildUpdateChecker(
	'https://example.com/',
	__FILE__,
	'example',
	6 //Check period in hours. The default is 12 hours.
);
```

You can force a check by calling the `checkForUpdates()` method:

```php
$myUpdateChecker->checkForUpdates();
```

If you need full control over when PUC checks for updates, one way to do that would be to set the check period to `0` (this disables periodic checks) and to call `checkForUpdates()` regularly instead. 

Alternatively, you could create a subclass of `Puc_v4p8_Plugin_UpdateChecker`, override the `createScheduler()` method and supply your own implementation of `Puc_v4p8_Scheduler`. However, there's not really any documentation on that.

**Comment by odil-io** [comment]
Thanks @YahnisElsts!

I shell check it out as soon as i have time for it and i'll report back my findings.

**Comment by odil-io** [comment]
Hi,

There wasn't a javascript error message that gave anything away. My guess that it worked fine, but because of the problem described below, PUC looked like it didn't work.

I found the problem, my releases where being released in a wrongfully matter.
First, the zip that was made for each theme had a version number lower than the actually newest release number. My automation process became faulty after I made some adjustments that apparently broke everything. (I automated trough gulp that the right files ware packaged in a WordPress compliant .zip and then automaticly uploaden to GitHub with a new tag)

What I'm now going to do is make a sort of boilerplate with minimal setup and try to include PUC in a reliable way so it's easy to keep up to date. I might even make a repo for that. For anyone reading this, don't expect anything soon.

Thank you Yahnis for your support and sorry for wasting your time!
